napari/components/_viewer/view/main.py

```diff
@@ -230,6 +239,24 @@ def on_draw(self, event):
         for layer in self.viewer.layers:
             layer.on_draw(event)
 
+    def dragEnterEvent(self, event):
+        if event.mimeData().hasUrls():
+            event.accept()
+        else:
+            event.ignore()
+
+    def dropEvent(self, event):
+        """Add an files from the drag and drop.
+        """
+        filenames = []
+        for url in event.mimeData().urls():
```

Comment:
Can you specify in a comment whether web URLs are allowed here? `skimage.io.imread` opens URLs so they totally could be!

Reply:
i've made it so we urls are now supported and added a comment. they work fine as along as its just an image you drag in, otherwise you can an `skimage.io.imread` error and it crashes. we'll have to work on error handling / testing etc. but that can come in a followup PR I think